app/models/katello/host/content_facet.rb

```diff
@@ -19,6 +20,7 @@ class ContentFacet < Katello::Model
 
       validates :content_view, :presence => true, :allow_blank => false
       validates :lifecycle_environment, :presence => true, :allow_blank => false
+      validates :content_source, :presence => true, :allow_blank => false
```

Comment:
we chatted briefly in IRC, but i'm not sure if this was before or after that.
There generally are two use cases:

1) A client is provisioned via foreman and katello.  As part of creating the new host, a user selects content source, content view, and lifecycle environment.  This content source selections tell foreman how to provision the host such that a) the install tree used for provisioning is from the smart proxy with content (previously known as a capsule), b) which capsule to register to after its provisioned 

2) if a client has already been provisioned (either because its been around for a while or the user uses something else for provisioning), the user pulls down the bootstrap from whichever smart proxy with content (SPWC, pronounced 'spwac') and registers manually.  In this case no smart proxy with content was ever selected.  

there is actually a 3rd use case where the user is using foreman/katello for provisioning but wants to use install media, but that may be a side issue that isn't all that important anyways.